### PR TITLE
Update omniauth and refactor

### PR DESCRIPTION
**Pull request by bbarberBPL**
- Updated to use  `omniauth ~> 2.1.0` 
- Removed `restclient` in favor of `http`
- Refactored `adaptor`  and `strategy` classes signifcantly 
- Added `rubocop` and linted 
- Dropped support for all ruby versions `< 2.6.10` 
- Made `activesupport < 7` a required dependency
- Updated `rspec` 
- Bumped version to 1.1.0

### CODE REVIEW

File: lib/omniauth/polaris/adaptor.rb [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+          access_id: @access_id
+        }
+
+        ensure_method!(@method)
+        @polaris_method = METHOD[@method]
+      end
+
+      def authenticate_patron(pin:, barcode:)
+        # According to the polaris api docs "Date must be within +/- 30 minutes of current time or request will fail"
+        http_date = 30.minutes.from_now.in_time_zone('GMT').strftime('%a, %d %b %Y %H:%M:%S %Z')
```
Comment (ebenenglish) [inline lib/omniauth/polaris/adaptor.rb]: Just curious, why do we need to set this to a future time? Why not just current time?

File: omniauth-polaris.gemspec [hunk 2]
```diff
@@ -1,29 +1,30 @@
...
+
+$:.push File.expand_path('lib', __dir__)
+require 'omniauth/polaris/version'
 
-  gem.required_ruby_version = '~> 2.4'
+Gem::Specification.new do |gem|
+  gem.name = 'omniauth-polaris'
+  gem.version = OmniAuth::Polaris::VERSION.dup
+  gem.authors = ['Steven Anderson', 'Ben Barber']
+  gem.email = ['sanderson@bpl.org', 'bbarber@bpl.org']
```
Comment (ebenenglish) [inline omniauth-polaris.gemspec]: Let's change Steven's email to `digital@bpl.org`

**Review by ebenenglish — COMMENTED** [review]
Looks great, just one question (for my own edification) and one minor change.

File: lib/omniauth/polaris/adaptor.rb (hunk 1 above)
Comment (bbarberBPL) [reply lib/omniauth/polaris/adaptor.rb]: Did you read the comment above that line?

File: omniauth-polaris.gemspec (hunk 2 above)
Comment (bbarberBPL) [reply omniauth-polaris.gemspec]: Do you want me to add you in there as well?

**Review by ebenenglish — APPROVED** [review]

**Pull request merged**